drivers/gpu/drm/tests/drm_framebuffer_test.c

```diff
@@ -366,8 +367,114 @@ static void drm_framebuffer_test_to_desc(const struct drm_framebuffer_test *t, c
 KUNIT_ARRAY_PARAM(drm_framebuffer_create, drm_framebuffer_create_cases,
 		  drm_framebuffer_test_to_desc);
 
+struct check_src_coords_case {
+	const int expect;
+	const uint32_t src_x;
+	const uint32_t src_y;
+	const uint32_t src_w;
+	const uint32_t src_h;
+	const struct drm_framebuffer fb;
+	const char *name;
+};
+
+static const struct check_src_coords_case check_src_coords_cases[] = {
```

Comment:
out of curiosity I've asked chat gpt for test cases https://chat.openai.com/share/9024132d-7b96-4e74-a342-5ae40b4b539b
not the best.


You need to use arbitrary non-zero values, but it's nice to use number close to the boundaries to catch off-by-one errors, so "source inside framebuffer" could be just big enough to fit.

I would also say to keep the arbitrary values consistent across test cases to facilitate the reader to identify what is changing between each case.

Since this function checks `src` _relative_ to `FB`, we can probably get away just modifying src values, that means keeping `.width = 800, .height = 600`, unless you see a reason to stressing one of them.

Reply:
Thanks for reviewing! I made a force push to try to address what you pointed out.

I made the tests be with minimal differences between them and replaced all arbitrary values to the higher one that "just fit". So, for example, in the tests where the source will be out of bound by the x coordinate, I leave the y related values untouched between the tests and with big enough values.

The only place where fb.{width,height} differs is in the tests that catch out of bound by y coordinates (they are just out of bound by x coordinates tests with swapped width/height and x/y values).